sdk/tests/tutorials/properties/test_derived_properties.py

```diff
@@ -0,0 +1,132 @@
+import unittest
+
+import json
+
+import lusid
+
+import lusid.models as models
+from utilities import InstrumentLoader
+from utilities import TestDataUtilities
+
+class DerivedPropertyTests(unittest.TestCase):
+
+
+    @classmethod
+    def setUpClass(cls):
+        # create a configured API client
+        api_client = TestDataUtilities.api_client()
+        cls.scopes_api = lusid.ScopesApi(api_client)
+        cls.property_definitions_api = lusid.PropertyDefinitionsApi(api_client)
+        cls.instruments_api = lusid.InstrumentsApi(api_client)
+        # load instruments from InstrumentLoader
+        instrument_loader = InstrumentLoader(cls.instruments_api)
+        cls.instrument_ids = instrument_loader.load_instruments()
+
+    def create_ratings_property(self, figi, fitch_value=None, moodys_value=None):
+
+        ratings = ["Fitch", "Moodys"]
+
+        for rating in ratings:
+            property_definition = models.CreatePropertyDefinitionRequest(
+                domain="Instrument",
+                scope="Test-Demo",
+                code="{}Rating".format(rating),
+                display_name="{}Rating".format(rating),
+                data_type_id=lusid.ResourceId(
+                    scope="system", code="number"))
+
+            try:
```

Comment:
Can you restructure the code, so the two properties get created once at the start?
In the current implementation, you attempt to create the properties each time you assign to a new instrument. 
This results in 6 calls to `create_property_definition` when only 2 are needed.

Reply:
Sorted as 2 separate functions. 

- def create_ratings_property(self, *ratings):
- def upsert_ratings_property(self, figi, fitch_value=None, moodys_value=None):

should only call `create_property_definition` by number of arguments passed.